src/app/clusters/window-covering-server/window-covering-server.cpp

```diff
@@ -163,46 +174,42 @@ namespace WindowCovering {
 
-    if (HasFeature(endpoint, WindowCoveringFeature::kLift) && HasFeature(endpoint, WindowCoveringFeature::kPositionAwareLift) &&
-        HasFeature(endpoint, WindowCoveringFeature::kAbsolutePosition) &&
-        EMBER_ZCL_STATUS_SUCCESS == Attributes::TargetPositionLiftPercent100ths::Get(endpoint, &liftPosition) &&
-        EMBER_ZCL_STATUS_SUCCESS == Attributes::InstalledOpenLimitLift::Get(endpoint, &liftLimit))
-    {
-        isOpen = liftPosition == liftLimit;
-    }
-    else if (HasFeature(endpoint, WindowCoveringFeature::kLift) &&
-             HasFeature(endpoint, WindowCoveringFeature::kPositionAwareLift) &&
-             EMBER_ZCL_STATUS_SUCCESS == Attributes::TargetPositionLiftPercent100ths::Get(endpoint, &liftPosition))
-    {
-        isOpen = 0 == liftPosition;
-    }
+    EmberAfStatus tiltStatus;
+    app::DataModel::Nullable<Percent100ths> tiltPosition;
 
-    if (HasFeature(endpoint, WindowCoveringFeature::kTilt) && HasFeature(endpoint, WindowCoveringFeature::kPositionAwareTilt) &&
-        HasFeature(endpoint, WindowCoveringFeature::kAbsolutePosition) &&
-        EMBER_ZCL_STATUS_SUCCESS == Attributes::TargetPositionTiltPercent100ths::Get(endpoint, &tiltPosition) &&
-        EMBER_ZCL_STATUS_SUCCESS == Attributes::InstalledOpenLimitTilt::Get(endpoint, &tiltLimit))
-    {
-        isOpen = isOpen && tiltPosition == tiltLimit;
-    }
-    else if (HasFeature(endpoint, WindowCoveringFeature::kTilt) &&
-             HasFeature(endpoint, WindowCoveringFeature::kPositionAwareTilt) &&
-             EMBER_ZCL_STATUS_SUCCESS == Attributes::TargetPositionTiltPercent100ths::Get(endpoint, &tiltPosition))
-    {
-        isOpen = isOpen && 0 == tiltPosition;
-    }
+    liftStatus = Attributes::TargetPositionLiftPercent100ths::Get(endpoint, liftPosition);
+    tiltStatus = Attributes::TargetPositionTiltPercent100ths::Get(endpoint, tiltPosition);
 
-    return isOpen;
+    if ((liftStatus != EMBER_ZCL_STATUS_SUCCESS) || liftPosition.IsNull())
+        return false;
+
+    if ((tiltStatus != EMBER_ZCL_STATUS_SUCCESS) || tiltPosition.IsNull())
+        return false;
+
+    return ((liftPosition.Value() == WC_PERCENT100THS_MIN) && (tiltPosition.Value() == WC_PERCENT100THS_MIN));
 }
 
 bool IsClosed(chip::EndpointId endpoint)
 {
-    return !IsOpen(endpoint);
+    EmberAfStatus liftStatus;
+    app::DataModel::Nullable<Percent100ths> liftPosition;
+
+    EmberAfStatus tiltStatus;
+    app::DataModel::Nullable<Percent100ths> tiltPosition;
+
+    liftStatus = Attributes::TargetPositionLiftPercent100ths::Get(endpoint, liftPosition);
+    tiltStatus = Attributes::TargetPositionTiltPercent100ths::Get(endpoint, tiltPosition);
+
+    if ((liftStatus != EMBER_ZCL_STATUS_SUCCESS) || liftPosition.IsNull())
+        return false;
+
+    if ((tiltStatus != EMBER_ZCL_STATUS_SUCCESS) || tiltPosition.IsNull())
+        return false;
+
+    return ((liftPosition.Value() == WC_PERCENT100THS_MAX) && (tiltPosition.Value() == WC_PERCENT100THS_MAX));
```

Comment:
Again, seems odd that this requires both to be supported.

Reply:
Same as above